dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNRatioTapChangerInterfaceIIDM.cpp

```diff
@@ -0,0 +1,146 @@
+//
+// Copyright (c) 2015-2020, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+// This file is part of Dynawo, an hybrid C++/Modelica open source time domain
+// simulation tool for power systems.
+//
+
+/**
+ * @file  DYNRatioTapChangerInterfaceIIDM.cpp
+ *
+ * @brief Ratio tap changer data interface : implementation file for IIDM implementation
+ *
+ */
+
+#include "DYNRatioTapChangerInterfaceIIDM.h"
+
+#include "DYNErrorQueue.h"
+#include "DYNMacrosMessage.h"
+#include "DYNStepInterfaceIIDM.h"
+// #include "DYNTrace.h"
+
+using boost::shared_ptr;
+
+namespace DYN {
+
+RatioTapChangerInterfaceIIDM::~RatioTapChangerInterfaceIIDM() {
+}
+
+RatioTapChangerInterfaceIIDM::RatioTapChangerInterfaceIIDM(powsybl::iidm::RatioTapChanger& tapChanger, const std::string& parentName) : tapChangerIIDM_(tapChanger) {
+  sanityCheck(parentName);
+}
+
+void
+RatioTapChangerInterfaceIIDM::addStep(const shared_ptr<StepInterface>& step) {
+  steps_.push_back(step);
+}
+
+std::vector<shared_ptr<StepInterface> >
+RatioTapChangerInterfaceIIDM::getSteps() const {
+  return steps_;
+}
+
+int
+RatioTapChangerInterfaceIIDM::getCurrentPosition() const {
+  return tapChangerIIDM_.getTapPosition();  // getTapPosition() is 'long' in powsybl
+}
+
+void
+RatioTapChangerInterfaceIIDM::setCurrentPosition(const int& position) {
+  tapChangerIIDM_.setTapPosition(position);
+}
+
+int
+RatioTapChangerInterfaceIIDM::getLowPosition() const {
+  return tapChangerIIDM_.getLowTapPosition();
+}
+
+unsigned int
+RatioTapChangerInterfaceIIDM::getNbTap() const {
+  return steps_.size();
+}
+
+bool
+RatioTapChangerInterfaceIIDM::hasLoadTapChangingCapabilities() const {
+  return tapChangerIIDM_.hasLoadTapChangingCapabilities();
+}
+
+bool
+RatioTapChangerInterfaceIIDM::getRegulating() const {
+  return tapChangerIIDM_.isRegulating();
+}
+
+double
+RatioTapChangerInterfaceIIDM::getTargetV() const {
+  if (std::isnan(tapChangerIIDM_.getTargetV())) {
+    return 99999.9;
+  }
+  return tapChangerIIDM_.getTargetV();
+}
+
+std::string
+RatioTapChangerInterfaceIIDM::getTerminalRefId() const {
+  return getRegulating() ? tapChangerIIDM_.getRegulationTerminal().get().getConnectable().get().getId() : "";
+}
+
+std::string
+RatioTapChangerInterfaceIIDM::getTerminalRefSide() const {
+  if (getRegulating()) {
+    /* DG- in powsybl-2 a terminal does not seem to have anymore 'side'. Only 3W-transformers have sides
+    switch (tapChangerIIDM_.terminalReference().side) {
```

Comment:
Did you ask to Thierry about this?

Reply:
Yes, he helped me a lot. In fact the 2WT derives from Branch which have 2 sides. The 3WT does not devives from branch and has t's own 3 sides. But we don't have a way to know if a RTC has a parent of that or that kind. There is a getParten() method in RTC which could lead to the information if it was not protected. Another way could be to had a member within ifce class, sayin which kind of parent it is, since ctor takes parent's name... I work on that besides working on PhaseTapChanger.

Reply:
Best will be to change the ctor of interface class to pass the transformer (2WT/3WT) as an argument. Doing so will will have access to sides and we will have the ParentName in case we need it for logging purpose. Now, without sanityCheck(), we don't need that name anymore, so we can change ctor already. Tell me if you want me to do that change.

Reply:
We could also exceptionally decide to modify the API. Let's keep this subject open and make a choice when at least one of the WT will be available

Reply:
Now I did not find a way to show side. We mentioned we can deliver now with "" for side, until we resolve that issue. Do you want a push just for updating coding style for RatioTapChanger ?